src/page-objects/base.ts

```diff
@@ -187,6 +187,15 @@ export default class BasePageObject {
     return elements[0].isDisplayed();
   }
 
+  async isDisplayedInViewport(selector: string) {
+    // browser.$ throws an error if element is not found, so we use this method to avoid try/catch
+    const elements = await this.browser.$$(selector);
```

Comment:
Do we really need to query all elements here when we only need one?

```js
const element = await this.browser.$(selector);

return (await element.isExisting()) && (await element. isDisplayedInViewport());
```

Reply:
Is this even a valid use-case? Do we ever call this check on a non-existing element?

I checked [the current usage](https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20isDisplayedInViewport&type=code) and I do not see it needed anywhere.

This code can be simplified